lib/filterx/filterx-scope.c

```diff
@@ -353,6 +355,11 @@ filterx_scope_make_writable(FilterXScope **pself)
 static void
 _free(FilterXScope *self)
 {
+  /* NOTE: update the number of inlined variable allocations */
+  gint variables_max = filterx_scope_variables_max;
+  if (variables_max < self->variables->len)
+    filterx_scope_variables_max = self->variables->len;
```

Comment:
Can we use `g_atomic_int` here?

I don't know any architectures where a `volatile` wouldn't be enough for this purpose, so I'm just asking.

Reply:
I would try to avoid atomics if possible, especially on the read side. It does not matter if we race on  the read itself, but atomics have additional overhead and we are allocating a scope 30-40 times per message.

By itself, one atomic is not visible in the profile, but the number of total atomic operations per second matters,
just as the reduction of malloc() calls improved scalability a lot.

So I would keep this as is, especially until our scalability starts to decline at around 5-6 cores.

With the current per core performance, we need to scale almost linearly on typical server class CPUs (at least 8
cores, but would be better to 16).